cairo/src/context.rs

```diff
@@ -223,10 +223,10 @@ impl Context {
     }
 
     #[doc(alias = "cairo_set_source")]
-    pub fn set_source(&self, source: &Pattern) -> Result<(), Error> {
-        source.status()?;
+    pub fn set_source(&self, source: impl AsRef<Pattern>) -> Result<(), Error> {
+        source.as_ref().status()?;
```

Comment:
Ideally you everywhere do something like `let source = source.as_ref()` in the beginning instead of calling it multiple times. Less line-noise

Reply:
that is true, will change that